Source/Evolve/WarpXEvolve.cpp

```diff
@@ -377,6 +376,18 @@ WarpX::OneStep_nosub (Real cur_time)
     // Deposit current j^{n+1/2}
     // Deposit charge density rho^{n}
 
+    if(istep[0]+1==end_fine_patch_step){
+        auto& warpx = WarpX::GetInstance();
+        SyncCurrent();
+        SyncRho();
+        const int coarse_lev = 0;
+        regrid(coarse_lev, cur_time);
+        mypc->Redistribute();
```

Comment:
see comment below.

I think particles are not moved from level 1 to level 0 here:
I think with the current logic: regrid then redistribute, the particles might remain on level 1?
https://github.com/AMReX-Codes/amrex/blob/development/Src/Particle/AMReX_ParticleContainerI.H#L1358-L1373

Reply:
update: according to a test case I did, it looks like particles move to level zero already.

Reply:
otherwise I would have had tested this diff:
```diff
diff --git a/Source/Evolve/WarpXEvolve.cpp b/Source/Evolve/WarpXEvolve.cpp
index c244fc1683..58ec676633 100644
--- a/Source/Evolve/WarpXEvolve.cpp
+++ b/Source/Evolve/WarpXEvolve.cpp
@@ -382,6 +382,7 @@ WarpX::OneStep_nosub (Real cur_time)
         SyncRho();
         const int coarse_lev = 0;
         regrid(coarse_lev, cur_time);
+        //mypc->moveParticlesToLevel0();
         mypc->Redistribute();
         Print() << "Remove the patch" << '\n';
         warpx.ComputeDt()  ;
diff --git a/Source/Particles/MultiParticleContainer.H b/Source/Particles/MultiParticleContainer.H
index 56f1b42ff3..840baac762 100644
--- a/Source/Particles/MultiParticleContainer.H
+++ b/Source/Particles/MultiParticleContainer.H
@@ -213,6 +213,8 @@ public:
 
     void Redistribute ();
 
+    void moveParticlesToLevel0 ();
+
     void defineAllParticleTiles ();
 
     void RedistributeLocal (const int num_ghost);
diff --git a/Source/Particles/MultiParticleContainer.cpp b/Source/Particles/MultiParticleContainer.cpp
index 156f90905d..252da4327a 100644
--- a/Source/Particles/MultiParticleContainer.cpp
+++ b/Source/Particles/MultiParticleContainer.cpp
@@ -571,6 +571,28 @@ MultiParticleContainer::Redistribute ()
 {
     for (auto& pc : allcontainers) {
         pc->Redistribute();
+        std::cout << "number of particles at level 0: " << pc->NumberOfParticlesAtLevel(0) << std::endl;
+        std::cout << "number of particles at level 1: " << pc->NumberOfParticlesAtLevel(1) << std::endl;
+    }
+}
+
+void
+MultiParticleContainer::moveParticlesToLevel0 ()
+{
+    int const old_finest_lvl = 1;
+    for (auto& pc : allcontainers) {
+        // copy
+        for (WarpXParIter pti(*pc, old_finest_lvl); pti.isValid(); ++pti)
+        {
+            int const new_lvl = 0;
+            int const ngrow = 0;
+
+            auto& tile = pc->ParticlesAt(old_finest_lvl, pti);
+            pc->AddParticlesAtLevel(tile, new_lvl, ngrow);
+        }
+
+        // remove
+        pc->RemoveParticlesAtLevel(old_finest_lvl);
     }
 }
```